code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,183 @@
+	return TRUE
+
+/datum/virology_goal/proc/check_for_duplicate()
+	return TRUE
+
+/proc/check_total_virology_goals_completion()
+	for(var/datum/virology_goal/V as anything in GLOB.virology_goals)
+		if(!V.completed)
+			return
+	GLOB.archived_virology_goals += GLOB.virology_goals
+	GLOB.virology_goals = list(new /datum/virology_goal/property_symptom, new /datum/virology_goal/virus, new /datum/virology_goal/virus/stealth)
+	for(var/obj/machinery/computer/pandemic/P in GLOB.pandemics)
+		P.print_goal_orders()
+
+/datum/virology_goal/Destroy()
+	GLOB.virology_goals -= src
+	GLOB.archived_virology_goals -= src
+	. = ..()
+
+/datum/virology_goal/Topic(href, href_list)
+	..()
+
+	if(!check_rights(R_EVENT))
+		return
+
+	if(href_list["remove"])
+		qdel(src)
+
+/datum/virology_goal/property_symptom
+	name = "Symptom With Properties Viral Sample Request"
+	var/goal_symptom //Type path of the symptom
+	var/goal_symptom_name
+	var/goal_property
+	var/goal_property_value
+
+/datum/virology_goal/property_symptom/New()
+	var/times_looped = 0 //The chance for this to make a infinite loop that lags the server is astronomically small but its still a chance
+	do
+		var/type = pick(subtypesof(/datum/symptom))
+		var/datum/symptom/S = new type()
+		goal_symptom = S.type
+		goal_symptom_name = S.name
+		goal_property = pick("resistance", "stealth", "stage rate", "transmittable")  
+		goal_property_value = rand(-18, 11)
+		switch(goal_property)
+			if("resistance")
+				goal_property_value += S.resistance
+			if("stealth")
+				goal_property_value += S.stealth
+			if("stage rate")
+				goal_property_value += S.stage_speed
+			if("transmittable")
+				goal_property_value += S.transmittable
+		qdel(S)
+		times_looped++
+	while(check_for_duplicate() && times_looped < 30)
+
+/datum/virology_goal/property_symptom/check_for_duplicate()
+	. = FALSE
+	if(!goal_symptom || !goal_property || !goal_property_value)
+		return
+	for(var/datum/virology_goal/property_symptom/V in (GLOB.archived_virology_goals + GLOB.virology_goals))
+		if(goal_symptom == V.goal_symptom && goal_property == V.goal_property && goal_property_value == V.goal_property_value)
+			return TRUE
+
+/datum/virology_goal/property_symptom/get_report(is_ui = FALSE)
+	var/return_text
+	if(!is_ui)
+		return_text += "<b>Effects of [goal_symptom_name] symptom and level [goal_property_value] [goal_property]</b><br>"
+	return_text += "Viral samples with a specific symptom and properties are required to study the effects of this symptom in various conditions. We need you to deliver [delivery_goal]u of viral samples containing the [goal_symptom_name] symptom and with the [goal_property] property at level [goal_property_value] along with 3 other symptoms to us through the cargo shuttle."
+	return return_text
+
+/datum/virology_goal/property_symptom/check_completion(list/datum/reagent/reagent_list)
+	. = FALSE
+	var/datum/reagent/blood/BL = locate() in reagent_list
+	if(!BL)
+		return
+	if(BL.data && BL.data["viruses"])
```

Comment:
Could you make this into an early return?

Reply:
isnt it a early return already?

Reply:
`if(BL.data && BL.data["viruses"])` can be rewritten into
```suggestion
	if(!BL.data || !BL.data["viruses"])
		return
```